cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/util/AwsStackNameCommonUtil.java

```diff
@@ -0,0 +1,34 @@
+package com.sequenceiq.cloudbreak.cloud.aws.common.util;
+
+import static org.slf4j.LoggerFactory.getLogger;
+
+import org.apache.commons.lang3.StringUtils;
+import org.slf4j.Logger;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+import com.google.common.base.Splitter;
+import com.sequenceiq.cloudbreak.cloud.context.AuthenticatedContext;
+
+@Component
+public class AwsStackNameCommonUtil {
+
+    private static final Logger LOGGER = getLogger(AwsStackNameCommonUtil.class);
+
+    @Value("${cb.max.aws.resource.name.length:}")
+    private int maxResourceNameLength;
+
+    public String getInstanceName(AuthenticatedContext ac, String groupName, Long privateId) {
+        if (StringUtils.isBlank(groupName)) {
+            throw new IllegalArgumentException("Group name cannot be empty, instance name cannot be generated");
+        }
+        String pattern = "%s-%s%s";
```

Comment:
Please extract this as a constant.

Reply:
I don't want to extract, I don't reuse it and we use the pattern mixed, also we use constant and variable

Reply:
done